http/e2e/e2e.go

```diff
@@ -0,0 +1,259 @@
+// Copyright 2023 Juan Pablo Tosso and the OWASP Coraza contributors
+// SPDX-License-Identifier: Apache-2.0
+
+package main
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"strings"
+	"time"
+)
+
+// External configurable variables:
+// - CORAZA_ENVOY: Interruptions at response body phase are allowed to return 200 (Instead of 403), but with a body full of null bytes. Defaults to "false".
+// - CORAZA_HOST: Main url used to perform requests. Defaults to "localhost:8080".
+// - HTTPBIN_HOST: Backend url, used for health checking reasons. Defaults to "localhost:8081".
+
+// Expected Coraza configs:
+/*
+// coraza.conf-recommended with SecRuleEngine On
+Include @demo-conf
+// Unmodified CRS configuration and rules
+Include @crs-setup-demo-conf
+Include @owasp_crs/*.conf
+// Custom rule for Coraza config check (ensuring that these configs are used)
+SecRule &REQUEST_HEADERS:coraza-e2e "@eq 0" "id:100,phase:1,deny,status:424,msg:'Coraza E2E - Missing header'"
+// Custom rules for e2e testing
+SecRule REQUEST_URI "@streq /admin" "id:101,phase:1,t:lowercase,deny"
+SecRule REQUEST_BODY "@rx maliciouspayload" "id:102,phase:2,t:lowercase,deny"
+SecRule RESPONSE_HEADERS::status "@rx 406" "id:103,phase:3,t:lowercase,deny"
+SecRule RESPONSE_BODY "@contains responsebodycode" "id:104,phase:4,t:lowercase,deny"
+*/
+
+const configCheckStatusCode = 424
+const healthCheckTimeout = 15 // Seconds
+
+func main() {
+	// Initialize variables
+	corazaEnvoy := false
+	corazaEnvoyString := os.Getenv("CORAZA_ENVOY")
+	if corazaEnvoyString == "true" {
+		corazaEnvoy = true
+	}
+	corazaHost := os.Getenv("CORAZA_HOST")
+	if corazaHost == "" {
+		corazaHost = "localhost:8080"
+	}
+	httpbinHost := os.Getenv("HTTPBIN_HOST")
+	if httpbinHost == "" {
+		httpbinHost = "localhost:8081"
+	}
+
+	healthEndPointUrl := "http://" + httpbinHost + "/status/200"
+	urlUnfiltered := "http://" + corazaHost
+	urlFiltered := urlUnfiltered + "/admin"
+	urlEcho := urlUnfiltered + "/anything"
+	urlFilteredRespHeader := urlUnfiltered + "/status/406"
+
+	healthChecks := []struct {
+		name         string
+		url          string
+		expectedCode int
+	}{
+		{
+			name:         "Backend health check",
+			url:          healthEndPointUrl,
+			expectedCode: 200,
+		},
+		{
+			name:         "Coraza health check",
+			url:          urlUnfiltered,
+			expectedCode: 200,
+		},
+		{
+			name:         "Coraza config check",
+			url:          urlUnfiltered,
+			expectedCode: configCheckStatusCode,
+		},
+	}
+
+	tests := []struct {
```

Comment:
We could move the tests outside the function.

Reply:
They are relying on variables that are created at runtime, moving the tests outside the function would require then some kind of translation with the right values 🧐